### PR TITLE
DATA: Source. Add Source Configurations

### LINKED ISSUE #39: DATA: Source - Add fluent configuration
## Model: Source

Properties:

| Field                  | Type                  | Description                                                |
| --- | --- | --- |
| Id                     | Guid                  | Our internal identifier                                    |
| Name | string (Required, max 255) | The name of the source system e.g. GitHub / DevOps |
| Url | string (Required, max 255) | The name of the source system e.g. GitHub / DevOps |

** **_Extra considerations: Add a unique constraint on Name. This will ensure we do not have duplicate sources._**

### PR DESCRIPTION
**Pull request by kandarpgautam**
closes #39 
- Added source configurations

### CODE REVIEW

File: GitFyle.Core.Api/Brokers/Storages/StorageBroker.Sources.Configuration.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright (c) The Standard Organization: A coalition of the Good-Hearted Engineers
+// ----------------------------------------------------------------------------------
+
+
```
Comment (glhays) [inline GitFyle.Core.Api/Brokers/Storages/StorageBroker.Sources.Configuration.cs:5]: Remove extra line

**Review by cjdutoit — APPROVED** [review]

File: GitFyle.Core.Api/Brokers/Storages/StorageBroker.Sources.Configuration.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+// ----------------------------------------------------------------------------------
+
+using GitFyle.Core.Api.Models.Foundations.Sources;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace GitFyle.Core.Api.Brokers.Storages
+{
+    internal partial class StorageBroker
+    {
+        void AddSourceConfigurations(EntityTypeBuilder<Source> builder)
```
Comment (cjdutoit) [inline GitFyle.Core.Api/Brokers/Storages/StorageBroker.Sources.Configuration.cs:12]: Add this in StorageBroker.cs and also add a migration for the unique constraint and length restrictions

```cs
        protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            AddSourceConfigurations(modelBuilder.Entity<Source>());
        }
```

**Review by cjdutoit — CHANGES_REQUESTED** [review]
Need migration and registration in StorageBroker

File: GitFyle.Core.Api/Brokers/Storages/StorageBroker.Sources.Configuration.cs (hunk 2 above)
Comment (kandarpgautam) [reply GitFyle.Core.Api/Brokers/Storages/StorageBroker.Sources.Configuration.cs:12]: currently, the OnModelCreating method is in the Contribution branch I guess. Adding it here will result In merge conflicts while waiting for it to be merged. and then we can add these configurations here.

Comment (cjdutoit) [reply GitFyle.Core.Api/Brokers/Storages/StorageBroker.Sources.Configuration.cs:12]: You are OK to add it here.  We will deal with potential conflicts when the other branches are merged.  We don't need to keep track of what is going on in other branches.  You only make sure that your branch is up to date with `main` any changes in other PR's is out of scope for this bit of work.

**Review by cjdutoit — APPROVED** [review]
Need migration and registration in StorageBroker

**Pull request merged**